lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -930,16 +932,35 @@ class OverrideFilteringConsumer : public VisibleDeclConsumer {
                           (BaseTy->getNominalOrBoundGenericNominal() ||
                            BaseTy->is<ArchetypeType>()) &&
                           VD->getDeclContext()->isTypeContext());
-      ModuleDecl *M = DC->getParentModule();
+
+      /// Substitue generic parameters in the signature of a found decl. The
+      /// returned type can be used to determine if we have already found a
+      /// conflicting declaration.
+      auto normalizeSignatureType = [&](CanType SignatureType, ValueDecl *VD,
+                                        Type BaseTy) -> CanType {
+        if (!SignatureType || !shouldSubst) {
+          return SignatureType;
+        }
+        if (auto GenFuncSignature =
+                SignatureType->getAs<GenericFunctionType>()) {
+          GenericEnvironment *GenEnv;
+          if (auto *GenCtx = VD->getAsGenericContext()) {
```

Comment:
I think it's always going to be a GenericContext if it has a GenericFunctionType

Reply:
It actually doesn’t. In the following example

```swift
protocol FooProtocol {
  var foo: Int { get set }
}

func test(foo: FooProtocol) {
  foo.#^COMPLETE^#
}
```

`foo` has a `SignatureType` of
```
(generic_function_type escaping
  (input=function_params num_params=1
    (param
      (generic_type_param_type depth=0 index=0)))
  (output=tuple_type num_elements=0)
  ( generic_sig=<τ_0_0 where τ_0_0 : FooProtocol>))
```

but the value decl `VD` (which is `foo`) doesn’t have a generic environment because it’s not generic..